### PR TITLE
[SPARK-7582] [MLLIB] user guide for StringIndexer

### PR DESCRIPTION
**Pull request by mengxr**
This PR adds a Java unit test and user guide for `StringIndexer`. I put it before `OneHotEncoder` because they are closely related. @jkbradley

### CODE REVIEW

File: docs/ml-features.md [hunk 1]
```diff
@@ -456,6 +456,122 @@ for expanded in polyDF.select("polyFeatures").take(3):
...
+<div data-lang="python" markdown="1">
+
+[`StringIndexer`](api/python/pyspark.ml.html#pyspark.ml.feature.StringIndexer) takes an input
+column name and an output column name.
+
+{% highlight python %}
+from pyspark.ml.feature import StringIndexer
+
+df = sqlContext.createDataFrame(
+    [(0, "a"), (1, "b"), (2, "c"), (3, "a"), (4, "a"), (5, "c")]
```
Comment (jkbradley) [inline docs/ml-features.md]: missing comma at end of line

**Comment by jkbradley** [comment]
Nice description!  LGTM, except for the missing comma

File: docs/ml-features.md (hunk 1 above)
Comment (mengxr) [inline docs/ml-features.md]: Thanks for catching it! I tested `VectorAssembler` example code but not this one ... this is why everything needs a test.

**Comment by mengxr** [comment]
test this please

**Comment by jkbradley** [comment]
test this please

**Comment by mengxr** [comment]
Merged into master and branch-1.4.

**Pull request closed without merging**